### PR TITLE
timer: Support set Tags for timer in timer framework

### LINKED ISSUE #44603: Support set `Tags` for timer in timer framework

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #44603

### What is changed and how it works?

Timer framework is a lib to provide timer functions that serval features can use. We can use `key` indicate different features but it is some what limited. So `Tags` can be used to class different timers in different dimissions.

For example if we have two timers used for GC and backup on time, their keys can be:

`/tidb/back` and `/tidb/gc`

For these above two timers, we can allow users to modify their meta through some other SQL syntax. So we can add tags `user_ctl` to them to mark it.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: timer/tablestore/store.go [hunk 1]
```diff
@@ -114,6 +119,11 @@ func (s *tableTimerStoreCore) List(ctx context.Context, cond api.Cond) ([]*api.T
 	}
 	defer back()
 
+	if sessVars := sctx.GetSessionVars(); sessVars.GetEnableIndexMerge() {
```
Comment (bb7133) [inline timer/tablestore/store.go:122]: Is there any plan we could remove this in the future? IMO it's quite confusing to the others.

Comment (lcwangchao) [reply timer/tablestore/store.go:122]: It's a limitation of multi-value index that is documented here: https://docs.pingcap.com/tidb/dev/choose-index#use-a-multi-valued-index . I think I can add some comments

Comment (bb7133) [reply timer/tablestore/store.go:122]: Got it, please add a comment about this.

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

File: timer/api/store.go [hunk 2]
```diff
@@ -106,6 +107,8 @@ type TimerCond struct {
 	// If `KeyPrefix is` true, it will check whether the timer's key is prefixed with `TimerCond.Key`.
 	// Otherwise, it will check whether the timer's key equals `TimerCond.Key`.
 	KeyPrefix bool
+	// Tags indicates to filter the timer record with specified tags
+	Tags OptionalVal[[]string]
```
[the remark this answers is not in the document]
Comment (lcwangchao) [reply timer/api/store.go:111]: Yes, it does not make a big difference but not all the same. One reason is to make sure that `TimerCond.FieldsSet` can return all set fields correctly...

**Pull request merged**